strawberry/channels/handlers/http_handler.py

```diff
@@ -170,19 +187,37 @@ def create_response(
             headers={k.encode(): v.encode() for k, v in sub_response.headers.items()},
         )
 
+    async def run(
+        self,
+        request: Any,
+        context: Optional[Any] = UNSET,
+        root_value: Optional[Any] = UNSET,
+    ) -> Union[ChannelsResponse, MultipartChannelsResponse]:
+        # putting this here just for type checking
+        raise NotImplementedError()
```

Comment:
This line is the cause of the `TimeoutError` errors for the Channels integrations tests as this is called from the `handle` method. I believe that a `Protocol` definition for this would be the correct way to define this type information? But Channels itself has no type information and pyright and mypy both will complain about the other unknown methods (e.g. `send_headers`) when I tested such a definition (without adding even more type definitions for the other unknown methods).

I believe this is also why I defined the return type directly in line [202](https://github.com/strawberry-graphql/strawberry/blob/fc7ecd8dfc7ca9a41da52d1183a5827b38d151fe/strawberry/channels/handlers/http_handler.py#L202) before.

Reply:
ouch! I'll remove it for now, thanks! I wish I we got an exception though :D